### PR TITLE
Speed up function template browsers in indirect analysis GUI

### PR DESCRIPTION
**Pull request by StephenSmith25**
This PR speeds up/ improves the responsiveness of the function browser interaction in the indirect analysis GUI. The unresponsiveness was noted while trying to implement default parameter estimates within the MSD tab and therefore warranted an investigation. The primary issues found were:

-	When a function was changed in the browser the parameters were updated, with each parameter update sending a Qt signal.
-	There are a lot of replot/redraw calls. For instance, changing the function in the browser requested 5 replots/redraws of the graph, despite the data not changing. Redrawing the graph in Matplotlib is a fairly costly procedure.

Some timing results of changing the function in the MSD tab (on my Debug build), after the first and second set of changes.

  | Without data loaded –time (ms) | With data loaded - time (ms)
-- | -- | --
Master | 587 | 1068
Sending Qt signals silently | 108 | 190
Removing excess redrawing | 50 | 88

**To test:**
- Data to test the GUI can be found in #26631
- Run through each tab, testing that the functionality still works.

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/scientific_interfaces/Indirect/IndirectFunctionBrowser/SingleFunctionTemplateBrowser.cpp [hunk 1]
```diff
@@ -173,6 +174,14 @@ void SingleFunctionTemplateBrowser::setParameterValue(
   m_parameterManager->setError(m_parameterMap[parameterName], parameterError);
 }
 
+void SingleFunctionTemplateBrowser::setParameterValueQuietly(
```
Comment (Mantid-Matthew) [inline qt/scientific_interfaces/Indirect/IndirectFunctionBrowser/SingleFunctionTemplateBrowser.cpp:177]: SingleFunctionTemplateBrowser has been updated here. Do we want to make the same updates to the ConvTemplateBrowser and IqtTemplateBrowser as well. They will presumably be having the same speed issues?

File: qt/scientific_interfaces/Indirect/IndirectFitPlotPresenter.cpp [hunk 2]
```diff
@@ -240,14 +251,21 @@ void IndirectFitPlotPresenter::setFitSingleSpectrumEnabled(bool enable) {
 }
 
 void IndirectFitPlotPresenter::updatePlots() {
+  HoldRedrawing holdRedrawing(m_view);
```
Comment (Mantid-Matthew) [inline qt/scientific_interfaces/Indirect/IndirectFitPlotPresenter.cpp:254]: Really like this implementation to stop multiple redraws but then guarantee a redraw when it falls out of scope.

File: qt/scientific_interfaces/Indirect/IndirectFunctionBrowser/SingleFunctionTemplateBrowser.cpp (hunk 1 above)
Comment (StephenSmith25) [reply qt/scientific_interfaces/Indirect/IndirectFunctionBrowser/SingleFunctionTemplateBrowser.cpp:177]: It appears those tabs are already sending the signals quietly, I only noticed the slow down on the MSD and F(q) tabs.

File: qt/scientific_interfaces/Indirect/IndirectFitAnalysisTab.cpp [hunk 3]
```diff
@@ -813,8 +813,7 @@ void IndirectFitAnalysisTab::respondToBackgroundChanged(double value) {
 
 void IndirectFitAnalysisTab::respondToFunctionChanged() {
   setModelFitFunction();
-  m_plotPresenter->updatePlots();
-  m_plotPresenter->updateGuessAvailability();
+  m_plotPresenter->updateFit();
```
Comment (Mantid-Matthew) [inline qt/scientific_interfaces/Indirect/IndirectFitAnalysisTab.cpp:816]: I think there is a slight change in functionality here. If you do a fit and then change the fit function the calculated fit line will remain unchanged whilst the plot guess will be updated. Don't know whether this is an issue or not so may be worth checking with scientists.

**Review by Matthew-Andrew — COMMENTED** [review]
Looks really good and at least for me GUI seems more responsive. A couple of minor comments but in general happy to approve if the scientists are happy with it.

File: qt/scientific_interfaces/Indirect/IndirectFitAnalysisTab.cpp (hunk 3 above)
Comment (SpencerHowells) [reply qt/scientific_interfaces/Indirect/IndirectFitAnalysisTab.cpp:816]: @StephenSmith25 @Mantid-Matthew If the fit function is changed I think it should remove the fitted plot.

Comment (StephenSmith25) [reply qt/scientific_interfaces/Indirect/IndirectFitAnalysisTab.cpp:816]: I thought the same, but as far as I could tell the previous behaviour was to keep the old fit on the graph until a new fit was obtained. In an effort to not make any functionality changes in this PR (to keep things small), I tried to replicate the existing behaviour. I can create a new issue to update the plot (i.e. remove the old fit) when the fit function is changed if desired.

Comment (SpencerHowells) [reply qt/scientific_interfaces/Indirect/IndirectFitAnalysisTab.cpp:816]: @StephenSmith25 ok. Leave as is for now

**Review by Matthew-Andrew — APPROVED** [review]
Changes look good to me and I see an improvement in GUI responsiveness. Also can't find any new issues introduced by these changes. Great work 👍

**Pull request merged**